packages/tools/src/stateManagement/annotation/AnnotationGroup.ts

```diff
@@ -0,0 +1,163 @@
+  }
+  /**
+   * Sets whether annotations belonging to this group are visible or not.
+   * If there are multiple groups, then the set visible false should be called
+   * before before re-enabling the other groups with setVisible true.
+   */
+  public setVisible(
+    isVisible = true,
+    baseEvent: BaseEventDetail,
+    filter?: (annotationUID: string) => boolean
+  ) {
+    if (this._isVisible === isVisible) {
+      return;
+    }
+    this._isVisible = isVisible;
+    this.annotationUIDs.forEach((uid) => {
+      const annotation = getAnnotation(uid);
+      if (!annotation) {
+        this.annotationUIDs.delete(uid);
+        return;
+      }
+      if (annotation.isVisible === isVisible) {
+        return;
+      }
+      if (!isVisible && filter?.(uid) === false) {
+        return;
+      }
+      annotation.isVisible = isVisible;
+      const eventDetail = {
+        ...baseEvent,
+        annotation,
+      };
+      triggerEvent(eventTarget, Events.ANNOTATION_MODIFIED, eventDetail);
+    });
+  }
+
+  public get isVisible() {
+    return this._isVisible;
+  }
+
+  public get isActive() {
+    return this._isActive;
+  }
+
+  /** Finds the nearby/next annotation in the given direction */
+  public findNearby(uid: string, direction: 1) {
+    if (!this._isVisible) {
+      return null;
+    }
+    const uids = [...this.annotationUIDs];
+    if (uids.length === 0) {
+      return null;
+    }
+    if (!uid) {
+      return uids[direction === 1 ? 0 : uids.length - 1];
+    }
+    const index = uids.indexOf(uid);
+    if (
+      index === -1 ||
+      index + direction < 0 ||
+      index + direction > uids.length
+    ) {
+      return null;
+    }
+    return uids[index + direction];
+  }
+
+  /**
+   * The active group is the group that gets automatically added to when annotations
+   * are added or removed.
+   * It is possible for multiple groups to be active, in which case each one will
+   * get added to on new or modified views.
+   * @param active
+   */
+  public setActive(active = true) {
+    this._isActive = active;
+  }
+
+  /**
+   * Adds the annotation to the group
+   * Does NOT change the visibility status of the annotation.
+   */
+  public add(annotationUID: string) {
+    this.annotationUIDs.add(annotationUID);
+  }
+
+  /**
+   * Removes the annotation from the group.
+   * Does not affect the visibility status of the annotation.
+   */
+  public remove(annotationUID: string) {
+    this.annotationUIDs.delete(annotationUID);
+  }
+
+  /**
+   * Adds listeners to automatically add/remove from this group when the group
+   * is active.
+   */
+  public addListeners() {
+    eventTarget.addEventListener(
+      Events.ANNOTATION_ADDED,
+      this.addAnnotationListener
+    );
+  }
+
+  /**
+   * Removes listeners that add to this group on creating a new instance.
+   */
+  public removeListeners() {
+    eventTarget.removeEventListener(
+      Events.ANNOTATION_ADDED,
+      this.addAnnotationListener
+    );
+  }
```

Comment:
These will cause so many side effects. I imagine there are use cases where you want a group, but you don't want adding or removing to be automated. Can we stick to explicit control over adding and removing to each group?

Reply:
And you get rid of the setActive too by moving to explicit. I think it is more simple and reduces potential unintended side effects